dcos/marathon.py

```diff
@@ -53,9 +53,21 @@ def _to_exception(response):
     """
 
     if response.status_code == 400:
-        return DCOSException(
-            'Error while fetching [{0}]: HTTP {1}: {2}'.format(
-                response.request.url, response.status_code, response.reason))
+        msg = 'Error on request [{0} {1}]: HTTP {2}: {3}'.format(
```

Comment:
This appears to be inconsistent in format with our existing error messaging. I tried adding a bad app definition to replicate this (and got a different error string to when I used the bad app definition you created):
```
➜  tmp  dcos marathon app add app.json
Error: Additional properties are not allowed ('command' was unexpected)
Value: {"container": {"docker": {"image": "superguenter/demo-app"}, "type": "DOCKER"}, "mem": 256, "cpus": 0.01, "ports": [3000], "instances": 2, "command": "rails server -p $PORT", "id": "rails-demo"}
➜  tmp  nano 400.json
➜  tmp  nano 400.json
➜  tmp  dcos marathon app add 400.json
Error on request [POST http://10.141.141.10:8080/v2/apps]: HTTP 400: Bad Request:
{
  "message": "Invalid JSON: JsResultException(errors:List((/container/docker/network,List(ValidationError(host is not a valid network type,WrappedArray())))))"
}
```

This is the app definition I used:
```json
{
  "container": {
    "type": "DOCKER",
    "docker": {
      "image": "superguenter/demo-app"
    }
  },
  "command": "rails server -p $PORT",
  "id": "rails-demo",
  "instances": 2,
  "cpus": 0.01,
  "mem": 256,
  "ports": [
    3000
  ]
}
```